src/main/java/org/radarcns/management/service/MetaTokenService.java

```diff
@@ -0,0 +1,145 @@
+import org.radarcns.management.service.dto.TokenDTO;
+import org.radarcns.management.web.rest.errors.RequestGoneException;
+import org.radarcns.management.web.rest.errors.NotFoundException;
+import org.radarcns.management.web.rest.errors.ErrorConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * Created by nivethika.
+ *
+ * <p>Service to delegate MetaToken handling.</p>
+ *
+ */
+@Service
+@Transactional
+public class MetaTokenService {
+
+    private final Logger log = LoggerFactory.getLogger(MetaTokenService.class);
+
+    @Autowired
+    private MetaTokenRepository metaTokenRepository;
+
+    @Autowired
+    private ManagementPortalProperties managementPortalProperties;
+
+    @Autowired
+    private SubjectService subjectService;
+
+    /**
+     * Save a metaToken.
+     *
+     * @param metaToken the entity to save
+     * @return the persisted entity
+     */
+    public MetaToken save(MetaToken metaToken) {
+        log.debug("Request to save MetaToken : {}", metaToken);
+        return metaTokenRepository.save(metaToken);
+    }
+
+    /**
+     * Get one project by id.
+     *
+     * @param tokenName the id of the entity
+     * @return the entity
+     */
+    public TokenDTO fetchToken(String tokenName) throws
+            MalformedURLException {
+        log.debug("Request to get Token : {}", tokenName);
+        MetaToken fetchedToken = getToken(tokenName);
+
+        // process the response if the token is not fetched or not expired
+        if (!fetchedToken.isFetched() && Instant.now().isBefore(fetchedToken.getExpiryDate())) {
+            // create response
+            TokenDTO result = new TokenDTO(fetchedToken.getToken(),
+                    new URL(managementPortalProperties.getCommon().getBaseUrl()),
+                    subjectService.getPrivacyPolicyUrl(fetchedToken.getSubject()));
+            // change fetched status to true.
+            fetchedToken.fetched(true);
+            save(fetchedToken);
+            return result;
+        } else {
+            throw new RequestGoneException("Token already fetched or expired. ",
+                META_TOKEN, "error.TokenCannotBeSent");
+        }
+
+
+    }
+
+    /**
+     * Gets a token from databased using the tokenName.
+     *
+     * @param tokenName tokenName.
+     * @return fetched token as {@link MetaToken}.
+     */
+    @Transactional(readOnly = true)
+    public MetaToken getToken(String tokenName) {
+        Optional<MetaToken> fetchedToken = metaTokenRepository.findOneByTokenName(tokenName);
+
+        if (fetchedToken.isPresent()) {
+            return  fetchedToken.get();
+        } else {
+            throw  new NotFoundException("Meta token not found with tokenName", META_TOKEN,
```

Comment:
There are two spaces between `throw` and `new`, and between `return` and `fetchedToken.get()` 😛